packages/experiments/src/implementation.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * wordpress/experimental – the utilities to enable private cross-package
+ * exports of experimental APIs.
+ *
+ * This "implementation.js" file is needed for the sake of the unit tests. It
+ * exports more than the public API of the package to aid in testing.
+ */
+
+/**
+ * The list of core modules allowed to opt-in to the experimental APIs.
+ */
+const CORE_MODULES_USING_EXPERIMENTS = [
+	'@wordpress/data',
+	'@wordpress/blocks',
+	'@wordpress/block-editor',
+];
```

Comment:
Why is there a fixed list of packages here?

Reply:
@youknowriad see my [other comment](https://github.com/WordPress/gutenberg/pull/46131#discussion_r1052298567) – this is an allowlist of which packages are allowed to access to `lock()` and `unlock()` functions.

Reply:
A bit weird though, if one of these package is not loaded on a given page, a third-party tool can still use the said API.

Reply:
There is no watertight solution here, unfortunately. Deterrents are as close as it gets: the existence of the `unlock()` API, the consent string, and the packages allowlist are designed to 1) make it impossible to use private APIs accidentally 2) communicate the consequences and 3) make your life harder if you still try to go for it.

Importantly, there's buy-in that this method should enable Gutenberg to smoothly ship and remove private cross-package experiments even after a major WordPress core release. It's hard to claim your project is broken because WordPress failed to keep the BC promise when your code literally says you knew that would happen.

Reply:
Let's keep the discussion on this particular point please. (For the buy in, I'd say it's not complete to say the least)

Reply:
On that particular point it's as I said – yes, you can work around it with enough effort. That effort, plus making your code even more likely to break, is the point.